krm-functions/lib/fn-runtime/design.md

```diff
@@ -0,0 +1,44 @@
+# fn-runtime
+
+The fn-runtime Project is a set of go libraries for building Kpt functions/controllers that are part of a "`Condition` choreography" also called a "`Condition` dance.
+
+The "`Condition` choreography/dance" is a set of kpt function and controllers that perform independent actions on a kpt package. Together and in harmony a declaritive outcome is achieved on a kpt package (a set of KRM resources).
+
+When comparing the conditional dance with a regular kubernetes controller built upon the controller runtime, we can make a number of analogies.
+- The event driven nature on a For object in the controller runtime is taken by porch, which periodically present the functions/controllers the latest status of the package. The fn/controllers provide an indepempodent operation on the package based on the resource presented.
+- The fn runtime presents For/Own/Watch resources that have a similar meaning in the fn-runtime as the controller runtime
+    - For: the parent krm resource the fn-runtime operates on
+    - Own: child resources which lifecycle is influenced by the parent resource or the parent resource depend upon
+    - Watch: extra resources relevant for the fn to operate on
+- Garbage collecion/owner reference is implemented in the fn-runtime using an annotation with owner key
+- Delete and finalizers are implemented using an annotiation and the delete operation is always performed by the downstream function.
```

Comment:
s/annotiation/annotation/

What does "delete operation is always performed by the downstream function" mean? Does this mean it has to use a lease renewal type of mechanism?